docs/rules/new-cap.md

```diff
@@ -76,6 +88,14 @@ Array of uppercase-starting function names that are permitted to be used without
 
 If provided, it must be an `Array`. The default values will continue to be excluded when `capIsNewExceptions` is provided.
 
+### `newIsCapWildcardExceptions`
+
+Array of lowercase library names where all child methods belong to are permitted to be used with `new` operator
```

Comment:
By granularity do you mean a separate option? my question [here](https://github.com/eslint/eslint/pull/5484#issuecomment-192981360) hopefully explains why I don't think it should be folded into the existing option.

As for the regex support - i don't think it's *needed*, but including it gets us a lot of functionality for free without having to implement anything special.